### ISSUE TITLE
Get-EventsFromEmailConfiguration - Error 500 - TxpConfiguration is not an SDS configuration

### ISSUE THREAD

**Issue by Matt-IgniteSolutions** [MicrosoftDocs/office-docs-powershell#10457]
Hi team,

I'm running into an issue when trying to use the EventsFromEmailConfiguration commands (both Get- and Set- produce the same error). I cannot find any information on the site about error messages or return codes that would assist in troubleshooting this. 

I am a Partner, and can log into multiple customer tenants via Delegated Admin. My delegated account is receiving the Global Admin role in the customer tenants as expected (as far as I can tell).

Issue description:

- In my tenant, as well as in any older customer tenants, I can log in via Powershell (connect-exchangeonline -userprincipalname matt@mydomain.com -delegatedorganization customerdomain.com), and run the Get-EventsFromEmailConfiguration and Set-EventsFromEmailConfiguration commands perfectly.
- In newer customer tenants, when I try this I receive the error copied in below. 
- If I then try to log into the customer tenant with a dedicated account within that customer tenant (IE, admin@customerdomain.com) that has the Global Admin role, I get the same error.  Both my delegated admin account and the dedicated customer admin account are unlicensed, if that makes any difference. 
- I've not found any way to be able to execute these commands on a newer customer tenant. 
- I'm not sure of the cut-off date, but when I try some of my older customers, it works fine. When I try and recently-created tenants, they fail the same way.
- I can confirm that I've run Enable-OrganizationCustomization in the newer tenants in case that was required, but it's now been a week since I ran that command and the same error exists.

Can anyone shed any light on why these same commands work in older tenants, but not in newer tenants? Are there any special requirements for the user account used to run these commands?

> PS C:\Users\MattRussell> get-eventsfromemailconfiguration user@customerdomain.com.au
Write-ErrorMessage : |Microsoft.Exchange.Management.Tasks.EventsFromEmailReadException|Error reading from
EventsFromEmail service : Error reading user settings from service: 500 {"Message":"An error has
occurred.","ExceptionMessage":"TxpConfiguration is not an SDS
configuration.","ExceptionType":"System.InvalidOperationException","StackTrace":"   at
Microsoft.Txp.Service.RestServer.BaseUserSettings.CopyFrom(ITxpConfiguration txpConfiguration)\r\n   at
Microsoft.TailoredExperiences.Core.TxpProcessorCommand.InternalExecuteWithAutoblocking(ITrackableOperation
fullCommand)\r\n   at Microsoft.TailoredExperiences.Core.TxpProcessorCommand.Execute()\r\n   at
Microsoft.TailoredExperiences.Core.TxpProcessor.GetUserSettings(IUserSettingsFactory userSettingsFactory, String
subScenarioName)\r\n   at Microsoft.Txp.Service.TxpCalendarEventController.c__DisplayClass52_0.b__0()\r\n   at
Microsoft.Txp.Service.BaseApiController.PerformOperation(String operationName, PerformOperationDelegate
operationDelegate, String userSmtp, String userId, TxpRequestTokenType txpRequestTokenType)\r\n   at
Microsoft.Txp.Service.TxpCalendarEventController.GetUserSettings(String userSmtp, ProtocolVersion protocolVersion)\r\n
  at lambda_method(Closure , Object , Object[] )\r\n   at
System.Web.Http.Controllers.ReflectedHttpActionDescriptor.ActionExecutor.c__DisplayClass10.b__9(Object instance,
Object[] methodParameters)\r\n   at
System.Web.Http.Controllers.ReflectedHttpActionDescriptor.ExecuteAsync(HttpControllerContext controllerContext,
IDictionary 2 arguments, CancellationToken cancellationToken)\r\n--- End of stack trace from previous location where
exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at
System.Web.Http.Controllers.ApiControllerActionInvoker.d__0.MoveNext()\r\n--- End of stack trace from previous
location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at
System.Web.Http.Controllers.ActionFilterResult.d__2.MoveNext()\r\n--- End of stack trace from previous location where
exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at
System.Web.Http.Dispatcher.HttpControllerDispatcher.d__1.MoveNext()"}.
At C:\Users\MattRussell\AppData\Local\Temp\tmpEXO_5hzbvyeq.11r\tmpEXO_5hzbvyeq.11r.psm1:1101 char:13
            Write-ErrorMessage $ErrorObject
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (user@customerdomain.com.au:String) [Get-EventsFromEmailConfiguration]
   , EventsFromEmailReadException
    + FullyQualifiedErrorId : [Server=SY4PR01MB7221,RequestId=1f850e32-2dd6-9a15-54e1-cad5191ef168,TimeStamp=Mon, 13 F
   eb 2023 22:32:17 GMT],Write-ErrorMessage 




>PS C:\Users\MattRussell> set-eventsfromemailconfiguration -identity user@customer.onmicrosoft.com -CreateEventsFromEmailAsPrivate $false
Write-ErrorMessage : |Microsoft.Exchange.Management.Tasks.EventsFromEmailWriteException|Error writing to
EventsFromEmail service : Error writing user settings from service: 500 {"Message":"An error has
occurred.","ExceptionMessage":"TxpConfiguration is not an SDS
configuration.","ExceptionType":"System.InvalidOperationException","StackTrace":"   at
Microsoft.Txp.Service.RestServer.BaseUserSettings.CopyTo(ITxpConfiguration txpConfiguration)\r\n   at
Microsoft.TailoredExperiences.Core.TxpProcessorCommand.InternalExecuteWithAutoblocking(ITrackableOperation
fullCommand)\r\n   at Microsoft.TailoredExperiences.Core.TxpProcessorCommand.Execute()\r\n   at
Microsoft.Txp.Service.TxpCalendarEventController.c__DisplayClass53_0.b__0()\r\n   at
Microsoft.Txp.Service.BaseApiController.PerformOperation(String operationName, PerformOperationDelegate
operationDelegate, String userSmtp, String userId, TxpRequestTokenType txpRequestTokenType)\r\n   at
Microsoft.Txp.Service.TxpCalendarEventController.SetUserSettings(IUserSettings userSettings, String userSmtp)\r\n   at
lambda_method(Closure , Object , Object[] )\r\n   at
System.Web.Http.Controllers.ReflectedHttpActionDescriptor.ActionExecutor.c__DisplayClass10.b__9(Object instance,
Object[] methodParameters)\r\n   at
System.Web.Http.Controllers.ReflectedHttpActionDescriptor.ExecuteAsync(HttpControllerContext controllerContext,
IDictionary 2 arguments, CancellationToken cancellationToken)\r\n--- End of stack trace from previous location where
exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at
System.Web.Http.Controllers.ApiControllerActionInvoker.d__0.MoveNext()\r\n--- End of stack trace from previous
location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at
System.Web.Http.Controllers.ActionFilterResult.d__2.MoveNext()\r\n--- End of stack trace from previous location where
exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at
System.Web.Http.Dispatcher.HttpControllerDispatcher.d__1.MoveNext()"}.
At C:\Users\MattRussell\AppData\Local\Temp\tmpEXO_5hzbvyeq.11r\tmpEXO_5hzbvyeq.11r.psm1:1101 char:13
              Write-ErrorMessage $ErrorObject 
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (user@customer.onmicrosoft.com:String) [Set-EventsFromEmailConfiguration]
   , EventsFromEmailWriteException
    + FullyQualifiedErrorId : [Server=SY4PR01MB7221,RequestId=2a3bdbbb-4612-49d1-18a4-187dec21329e,TimeStamp=Mon, 13 F
   eb 2023 22:44:29 GMT],Write-ErrorMessage 

And when I try it within my own tenant, signed in with my same global admin account:

> PS C:\Users\MattRussell> get-eventsfromemailconfiguration -identity matt@mydomain.com.au
>
>
>DisableReason                    : None
>CreateEventsFromEmailAsPrivate   : True
>EntityTypeProcessorLevelSettings : {Invoice, RentalCarReservation, EventReservation, FoodEstablishmentReservation...}
>Identity                         :
>IsValid                          : True
>ObjectState                      : New
>
>WARNING: The following properties are deprecated: EventReservationProcessingLevel,
>FoodEstablishmentReservationProcessingLevel, InvoiceProcessingLevel, ServiceReservationProcessingLevel

And when I run the same command whilst logged in with my delegated admin account into an old customer tenant:
>PS C:\Users\MattRussell> get-eventsfromemailconfiguration -identity person@customerdomain.com.au
>
>
>DisableReason                    : None
>CreateEventsFromEmailAsPrivate   : True
>EntityTypeProcessorLevelSettings : {Invoice, RentalCarReservation, EventReservation, FoodEstablishmentReservation...}
>Identity                         :
>IsValid                          : True
>ObjectState                      : New
>
>WARNING: The following properties are deprecated: EventReservationProcessingLevel,
>FoodEstablishmentReservationProcessingLevel, InvoiceProcessingLevel, ServiceReservationProcessingLevel

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: f69b70b6-3227-b093-ede4-65af97afd8a8
* Version Independent ID: 6e88af7f-084f-871a-ff3b-6e7307f4f199
* Content: [Get-EventsFromEmailConfiguration (ExchangePowerShell)](https://learn.microsoft.com/en-us/powershell/module/exchange/get-eventsfromemailconfiguration?view=exchange-ps)
* Content Source: [exchange/exchange-ps/exchange/Get-EventsFromEmailConfiguration.md](https://github.com/MicrosoftDocs/office-docs-powershell/blob/main/exchange/exchange-ps/exchange/Get-EventsFromEmailConfiguration.md)
* Service: **exchange-powershell**
* GitHub Login: @chrisda
* Microsoft Alias: **chrisda**

**Comment by chrisda** [comment]
@Matt-IgniteSolutions, you should open a support ticket about this.

**Comment by Matt-IgniteSolutions** [comment]
Thanks Chris.

I did open a ticket, but Support told me "We do not support scripting here in this support department. Its "best effort" but it's out of scope.". That's when I came back to review the documentation to see if there was anything listed here about error codes or failure modes or even what an "SDS Configuration" is. 

That's OK - I figured it was a long shot (and not really the right place to ask), but was worth a shot. Happy for you to close/delete this thread since it's not a documentation question. Thanks for getting back to me. 

Cheers,
Matt
